apiclient/helpers.go

```diff
@@ -314,5 +320,49 @@ func UnmarshalFaucetPackage(data []byte) (*models.FaucetPackage, error) {
 		Payload:   fpackage.FaucetPayload,
 		Signature: fpackage.Signature,
 	}, nil
+}
 
+// GenerateSik function generates the Secret Identity Key for the current client
+// account with the signature and the secret provided following the definition:
+//
+//	SIK = poseidon(address, signature, secret*)
+//
+// *The secret is optional.
+func (c *HTTPclient) GenerateSik(sign, secret []byte) ([]byte, error) {
```

Comment:
I don't think we need this HTTPclient method. 

The SIK generation should be only part of the crypto/ethereum package. Let's abstract it.